### ISSUE TITLE
Document ocaml_compiler_internal_params

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7311]
**Original bug ID:** 7311
**Reporter:** @gasche
**Assigned to:** @damiendoligez
**Status:** closed (set by @damiendoligez on 2016-11-03T13:25:08Z)
**Resolution:** fixed
**Priority:** normal
**Severity:** text
**Version:** 4.03.0
**Target version:** 4.04.0 +dev / +beta1 / +beta2
**Fixed in version:** 4.04.0 +dev / +beta1 / +beta2
**Category:** documentation
**Related to:** #7007

## Bug description

The file ocaml_compiler_internal_params introduced by 

  https://github.com/ocaml/ocaml/pull/428

is undocumented. Not knowing about it can make it difficult for users to properly debug an OCaml environment to understand where some options are coming from.

## Additional information

Documentation should be added in the chapter "Batch compilation (ocamlc)", section "Options" (manual/manual/cmds/comp.etex), and also in "Native-code compilation" (manual/manual/cmds/native.etex).

**Comment by vicuna** [comment]
**Comment author:** @gasche

I think it is a mistake to delay this PR ad infinitum. People introducing new features should document them. The documentation should go in the release introducing the feature, and we already slipped that (it was added in 4.03). I'm moving this back to 4.04+dev to make sure the issue is properly discussed/considered -- but if there is strong consensus to delay I won't oppose.
